atomic_reactor/utils/operator.py

```diff
@@ -25,6 +27,136 @@
 OPERATOR_CSV_KIND = "ClusterServiceVersion"
 
 
+def is_dict(obj):
+    """
+    Check if object is a dict or the ruamel.yaml equivalent of a dict
+    """
+    return isinstance(obj, (dict, CommentedMap))
+
+
+def is_list(obj):
+    """
+    Check if object is a list or the ruamel.yaml equivalent of a list
+    """
+    return isinstance(obj, (list, CommentedSeq))
+
+
+def is_str(obj):
+    """
+    Check if object is a string or bytes. On python 3, checking for string
+    would be sufficient, but on python 2, it may not be.
+    """
+    return isinstance(obj, (bytes, six.text_type))
+
+
+class PullspecRegex(object):
+    """
+    Regular expressions for all things related to pullspecs
+    """
+
+    # Alphanumeric characters
+    _alnum = r"[a-zA-Z0-9]"
+    # Characters that you might find in a registry, namespace, repo or tag
+    _name = r"[a-zA-Z0-9\-._]"
+    # Base16 characters
+    _base16 = r"[a-fA-F0-9]"
+
+    # A basic name is anything that contains only alphanumeric and name
+    # characters and starts and ends with an alphanumeric character
+    _basic_name = r"(?:(?:{alnum}{name}*{alnum})|{alnum})".format(alnum=_alnum, name=_name)
+
+    # A basic tag is ':' followed by a basic name
+    _basic_tag = r"(?::{basic_name})".format(basic_name=_basic_name)
+    # A digest is "@sha256:" followed by base16 characters
+    _digest = r"(?:@sha256:{base16}+)".format(base16=_base16)
+
+    # A tag is either a basic tag or a digest
+    _tag = r"(?:{basic_tag}|{digest})".format(basic_tag=_basic_tag, digest=_digest)
+
+    # Registry is a basic name that contains at least one dot
+    # followed by an optional port number
+    _registry = r"(?:{alnum}{name}*\.{name}*{alnum}(?::\d+)?)".format(alnum=_alnum, name=_name)
+
+    # Namespace is a basic name
+    _namespace = _basic_name
+
+    # Repo is a basic name followed by a tag
+    _repo = _basic_name + _tag
+
+    # NOTE: Tag is REQUIRED, otherwise regex picks up too many false positives,
```

Comment:
This comment is important but it seems out of place. Can we move it up to where `_repo` is defined?